tests/functional/modules/test_zos_zfs_resize_func.py

```diff
@@ -0,0 +1,551 @@
+    for result in results.contacted.values():
+        assert result.get('target') == ds_name
+        assert result.get('size') == size
+        assert result.get('rc') == 1
+        assert result.get('changed') is False
+        assert result.get('msg') == "ZFS Target {0} does not exist".format(ds_name)
+
+def test_mount_point_does_not_exist(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 2500
+    try:
+        hosts.all.zos_data_set(name=ds_name, type="zfs", space_primary=1, space_type="m")
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=size,)
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('size') == size
+            assert result.get('rc') == 1
+            assert result.get('changed') is False
+            assert  "No mount points were found in the following output:" in result.get('msg')
+    finally:
+        hosts.all.zos_data_set(name=ds_name, state="absent")
+
+def test_no_operation_executed(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 1440
+    try:
+        mount_folder = set_environment(ansible_zos_module=hosts, ds_name=ds_name)
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=size,)
+        for result in results.contacted.values():
+            assert result.get("changed") == False
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 0
+            assert result.get('size') == size
+            assert result.get('space_type') == "K"
+            assert result.get('stdout') == "Same size as size of the file {0}".format(ds_name)
+            assert result.get('new_size') >= result.get('old_size')
+            assert result.get('new_free') >= result.get('old_free')
+    finally:
+        clean_up_environment(hosts=hosts, ds_name=ds_name, temp_dir_name=mount_folder)
+
+def test_no_space_to_operate(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 100
+    try:
+        mount_folder = set_environment(ansible_zos_module=hosts, ds_name=ds_name)
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=size,)
+        for result in results.contacted.values():
+            assert result.get("changed") == False
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 1
+            assert result.get('size') == size
+            assert result.get('space_type') == "K"
+            assert result.get('msg') == "Not enough space to shrink"
```

Comment:
we can maybe change the response to "Not enough free space in the filesystem to shrink."